roles/k3s/master/tasks/main.yml

```diff
@@ -153,13 +152,20 @@
     owner: "{{ ansible_user }}"
     mode: "u=rw,g=,o="
 
-- name: Configure kubectl cluster to https://{{ apiserver_endpoint | ansible.utils.ipwrap }}:6443
+- name: Configure kubectl cluster to https://{{ apiserver_endpoint }}:6443
   command: >-
     k3s kubectl config set-cluster default
-      --server=https://{{ apiserver_endpoint | ansible.utils.ipwrap }}:6443
+      --server=https://{{ apiserver_endpoint }}:6443
       --kubeconfig ~{{ ansible_user }}/.kube/config
   changed_when: true
 
+- name: Fetch cluster config file from first master to push to agent nodes - allowing agent nodes to run k3s kubectl xxx (patching, convenience)
+  ansible.builtin.fetch:
+    src: ~{{ ansible_user }}/.kube/config
+    flat: true
+    dest: /tmp/kube_config.{{ apiserver_endpoint }}
+  when: ansible_hostname == hostvars[groups['master'][0]]['ansible_hostname']
```

Comment:
Could we get around this intermediate step by delegating the `copy` task for the `node` role to the first master?

Reply:
technically, probably, yes.  Drawbacks to that -- at least as I see it.  I'm running ansible from my mac, across to 4 RPIs.  I didn't put my ansible user's ssh key in each node, rather, just from my mac.  I think this would be "harder" for users to get started by requiring ssh keys on the master node too...

Reply:
oh, and I didn't delete the file, as I need it locally, eventually, anyway...  my post-process is to 'rename' the default user and default cluster name so that I can distinguish between two different clusters today.